internal/api/v1/organizations.go

```diff
@@ -93,3 +96,73 @@ func (oc OrganizationsController) Create(w http.ResponseWriter, r *http.Request)
 	w.WriteHeader(http.StatusCreated)
 	w.Write([]byte{})
 }
+
+func (oc OrganizationsController) Delete(w http.ResponseWriter, r *http.Request) {
+	params := httprouter.ParamsFromContext(r.Context())
+	org := params.ByName("org")
+
+	gitea, err := gitea.New()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	cluster, err := kubernetes.GetCluster()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	exists, err := organizations.Exists(cluster, org)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+	if !exists {
+		http.Error(w, fmt.Sprintf("Organization '%s' does not exist", org),
+			http.StatusNotFound)
+		return
+	}
+
+	apps, err := application.List(cluster, org)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	servicesToBeDeleted := []string{}
+
+	for _, app := range apps {
```

Comment:
I wonder if we gain anything by deleting things manually. If everything we delete lives inside the org's namespace, can't we simply delete the namespace and wait until it's gone? Is there anything we need to delete that won't be deleted this way?

Reply:
Yes, @thardeck suggested that. I thought we should delete things and then delete the namespace since this is nice way step by step to do it. Sometimes I see terminating namespace for more than 20 min waiting for some resources/finalizers ... By this, we can be sure, that in the future such problems don't occur or are easy to identify.

Reply:
True, but this could be really slow if you had, let's say, 1000 apps running in an org. Even if we manually delete things, we should probably do it in parallel, unless there is a reason for some actions to happen sequentially. Feel free to improve this on a new PR, since this is optimization.

Reply:
How would we test this (and scaling in general) ?

Reply:
Do you mean one-off test or as part of the test suite? As a test suite I think this has to be a new one (a performance test suite or something). Deploying 1000 apps may need us to take some shortcuts to avoid staging etc. As an one off it should be relatively easy.

Generally, scaling involves more concepts like HA-ability of components, pod-affinity rules [and others](https://opsani.com/blog/10-kubernetes-performance-tuning-tips-best-practices/). We haven't even started working on optimizations like this since we are at the `make it work` stage yet. That's why I think we could postpone fixing my comment for another (later) PR. Unless  it bites us earlier (e.g. make tests or development slow).

Reply:
I would recommend a smaller-scope issue. Just to delete orgs by deleting objects in parallel. Otherwise it will be too open ended.

Reply:
Done.